logisland-components/logisland-processors/logisland-processor-computer-vision/src/main/java/com/hurence/logisland/cv/processor/RunScript.java

```diff
@@ -0,0 +1,250 @@
+        return Collections.unmodifiableList(descriptors);
+    }
+
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext context) {
+        final List<ValidationResult> validationResults = new ArrayList<>(super.customValidate(context));
+
+        logger.debug("customValidate");
+
+       /* if (context.getPropertyValue(PROCESSING_MODE).isSet() &&
+                context.getPropertyValue(PROCESSING_MODE).asString().equals(RECORD_BASED.getValue())) {
+
+            if (context.getPropertyValue(INPUT_FIELD).isSet()) {
+                // attempt to use both modes -> error
+                validationResults.add(
+                        new ValidationResult.Builder()
+                                .explanation("You must declare " + PROCESSING_MODE.getName() + " or " + INPUT_FIELD.getName() + " but not both")
+                                .valid(false)
+                                .build());
+            }
+
+            if (context.getPropertyValue(OUTPUT_FIELD).isSet()) {
+                // attempt to use both modes -> error
+                validationResults.add(
+                        new ValidationResult.Builder()
+                                .explanation("You must declare " + PROCESSING_MODE.getName() + " or " + OUTPUT_FIELD.getName() + " but not both")
+                                .valid(false)
+                                .build());
+            }
+        }*/
+
+        return validationResults;
+    }
+
+
+    @Override
+    public void init(final ProcessContext context) throws InitializationException {
+        super.init(context);
+
+        try {
+            NativeLoader.loadLibrary(Core.NATIVE_LIBRARY_NAME);
+
+            RT.load("clojure/core");
+            StringReader stringReader = new StringReader(context.getPropertyValue(SCRIPT_CODE).asString());
+
+            // Load the Clojure script -- as a side effect this initializes the runtime.
+            Compiler.load(stringReader);
+
+            processFunction = RT.var(context.getPropertyValue(SCRIPT_NS).asString(),
+                    context.getPropertyValue(SCRIPT_FUNCTION).asString());
+
+            isInitialized = true;
+        } catch (IOException | ClassNotFoundException e) {
+            throw new InitializationException(e);
+        }
+
+
+    }
+
+
+    @Override
+    public Collection<Record> process(ProcessContext context, Collection<Record> records) {
+        if (!isInitialized) {
+            logger.error("Processor not initialized, returning input records and doing nothing!");
+            return records;
+        }
+
+        String imageInputField = context.getPropertyValue(INPUT_FIELD).asString();
+        String imageOutputField = context.getPropertyValue(OUTPUT_FIELD).asString();
+        String imageFormat = context.getPropertyValue(IMAGE_FORMAT).asString();
+
+        // do we replace existing records or add new ones ?
+        Collection<Record> outputRecords = new ArrayList<>(records);
```

Comment:
why cloning records first to eventually discard this afterwards ? This is not efficient, I would do the opposite.

Reply:
totally right man !